caproto/_data.py

```diff
@@ -869,18 +909,25 @@ async def write_metadata(self, publish=True, units=None, precision=None,
                     pass
                 data[kw] = value
 
-        if any(alarm_val is not None
-               for alarm_val in (status, severity)):
+        if timestamp is not None:
+            self._data["timestamp"] = TimeStamp.from_flexible_value(timestamp)
+
+        if status is not None or severity is not None:
             await self.alarm.write(status=status, severity=severity,
                                    publish=publish)
 
         if publish:
             await self.publish(SubscriptionType.DBE_PROPERTY)
 
     @property
-    def epics_timestamp(self):
-        'EPICS timestamp as (seconds, nanoseconds) since EPICS epoch'
-        return timestamp_to_epics(self._data['timestamp'])
+    def timestamp(self) -> float:
+        """UNIX timestamp in seconds."""
+        return self._data["timestamp"].timestamp
+
+    @property
+    def epics_timestamp(self) -> TimeStamp:
+        """EPICS timestamp as (seconds, nanoseconds) since EPICS epoch."""
+        return copy.copy(self._data["timestamp"])
```

Comment:
Part of me wants to suggest that it's fine to return the original here --- if you modify that struct you deserve what happens to you --- but you are probably right about this.

Reply:
I'm 50/50 on it as well, and not to mention lazy. So I think the current implementation wins out!